### PR TITLE
Added maximum non-adjacent sum

### PR DESCRIPTION
**Pull request by ruppysuppy**
### **Describe your change:**

* [X] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by ruppysuppy** [comment]
Could someone help? The build is failing due to [ciphers/affine_cipher.py](https://github.com/TheAlgorithms/Python/blob/master/ciphers/affine_cipher.py#L16). I went through the logs at Travis CI

File: dynamic_programming/max_non_adjacent_sum.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from typing import List
```
Comment (mateuszz0000) [inline dynamic_programming/max_non_adjacent_sum.py:1]: Could you provide some source for newcomers who want to learn theory?

File: dynamic_programming/max_non_adjacent_sum.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    if (len(nums) == 0):
+        return 0
+
+    max_including = nums[0]
+    max_excluding = 0
+
+    for num in nums[1:]:
+        temp = max_including
+        max_including = max_excluding + num
+        max_excluding = max(temp, max_excluding)
```
Comment (cclauss) [inline dynamic_programming/max_non_adjacent_sum.py:44]:
```suggestion
        max_including, max_excluding = max_excluding + num, max(temp, max_excluding)
```
Python has tuple assignment that allows you to sidestep the need for temporary variables.

File: dynamic_programming/max_non_adjacent_sum.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    18
+
+    >>> max_non_adj_sum([-1, -5, -3, -7, -2, -2, -6])
+    0
+
+    >>> max_non_adj_sum([499, 500, -3, -7, -2, -2, -6])
+    500
+    '''
+
+    if (len(nums) == 0):
```
[the remark this answers is not in the document]
Comment (cclauss) [reply dynamic_programming/max_non_adjacent_sum.py:35]: My favorite is to simplify to `if not nums:`

**Comment by cclauss** [comment]
Do not worry about bugs in files that you did not touch.  Your branch is just a bit behind master.

File: dynamic_programming/max_non_adjacent_sum.py (hunk 2 above)
Comment (mateuszz0000) [reply dynamic_programming/max_non_adjacent_sum.py:44]: ahh @cclauss always first :cry:

File: dynamic_programming/max_non_adjacent_sum.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+
+def max_non_adj_sum(nums: List[int]) -> int:
```
Comment (cclauss) [inline dynamic_programming/max_non_adjacent_sum.py:4]:
```suggestion
def maximum_non_adjacent_sum(nums: List[int]) -> int:
```

File: dynamic_programming/max_non_adjacent_sum.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+    Parameters
+    ----------
+    nums : List[int]
+        List of integers for the maximum non-adjacent sum computation
+
+    Returns
+    -------
+    int
+        maximum non-adjacent sum
```
Comment (cclauss) [inline dynamic_programming/max_non_adjacent_sum.py:16]:
```suggestion
    Find the maximum non-adjacent sum of the integers in the nums input list
```
Type hints and careful naming of functions and variables makes a lot of comments unnecessary boilerplate.
1. `def` tells us that this is a function.
2. The self-documenting function name tells us what this function does.
3. The type hints already tell us that `nums` is a list of integers.
4. The type hint on the return value tells us that we will return an it.

Comment (ruppysuppy) [reply dynamic_programming/max_non_adjacent_sum.py:18]: Looking into it

**Pull request merged**